functors_task_2/student/include/impl.h

```diff
@@ -76,8 +81,19 @@ class DataBrowser
     template<class Functional, typename Output>
     bool invokeDataRequest(Functional method, const std::unique_ptr<IDataSelector>& selector, Output& result) const
     {
-        // adapt function
-        // call selector member
+        if (selector)
+        {
+            return std::ref(method)(selector.get(), result);
```

Comment:
please leave the "normal" execution flow till the end, and check the preconditions
so, swap line 86 with line 89